### PR TITLE
[after #697] feat: api v1 contract info

### PR DESCRIPTION
**Pull request by songwongtp**

### CODE REVIEW

File: src/lib/types/projects.ts [hunk 1]
```diff
@@ -133,3 +133,17 @@ export const zPublicAccountInfo = z.object({
 });
 
 export type PublicAccountInfo = z.infer<typeof zPublicAccountInfo>;
+
+export const zPublicContractInfo = z.object({
+  address: zContractAddr,
+  admin: zAddr,
+  code: z.number(),
```
Comment (evilpeach) [inline src/lib/types/projects.ts:140]: codeId? If yes, should it be non negative?

File: src/lib/services/account.ts [hunk 2]
```diff
@@ -9,27 +9,27 @@ const zIcns = z.object({
...
     public_info: zPublicAccountInfo.nullable(),
   })
-  .transform((accountInfo) => ({
-    icns: accountInfo.icns,
-    projectInfo: accountInfo.project_info,
-    publicInfo: accountInfo.public_info,
+  .transform((accountData) => ({
+    icns: accountData.icns,
+    projectInfo: accountData.project_info,
+    publicInfo: accountData.public_info,
```
Comment (evilpeach) [inline src/lib/services/account.ts:21]: Please use `snakeToCamel`

File: src/lib/services/contract.ts [hunk 3]
```diff
@@ -132,6 +121,94 @@ export const getAdminContractsByAddress = async (
...
+    createdHeight: value.created_height,
+    createdTimestamp: value.created_timestamp,
+    cw2Contract: value.cw2_contract,
+    cw2Version: value.cw2_version,
+    initMsg: value.init_msg,
+    initProposalId: value.init_proposal_id,
+    initProposalTitle: value.init_proposal_title,
+    initTxHash: value.init_tx_hash,
+    instantiator: value.instantiator,
+    label: value.label,
```
Comment (evilpeach) [inline src/lib/services/contract.ts:158]: Can we use snakeToCamel as well?

File: src/lib/services/contract.ts [hunk 4]
```diff
@@ -132,6 +121,94 @@ export const getAdminContractsByAddress = async (
...
+export type Contract = z.infer<typeof zContract>;
+
+const zContractData = z
+  .object({
+    project_info: zProjectInfo.nullable(),
+    public_info: zPublicContractInfo.nullable(),
+    contract: zContract.nullable(),
+    contract_rest: zContractRest.nullable(),
+  })
+  .transform((value) => ({
```
Comment (evilpeach) [inline src/lib/services/contract.ts:169]: ditto

File: src/lib/services/contract.ts (hunk 3 above)
Comment (songwongtp) [reply src/lib/services/contract.ts:158]: need to preserve fields in `contract_rest`

**Review by evilpeach — APPROVED** [review]

**Pull request merged**